File: vimeo-networking/src/main/java/com/vimeo/networking/model/RemoveVideoFromAlbum.java
package com.vimeo.networking.model;

import com.google.gson.annotations.SerializedName;

import org.jetbrains.annotations.NotNull;

import java.io.Serializable;

/**
 * An object that is used to patch video addition updates to an Album.
 */
public class RemoveVideoFromAlbum implements Serializable {

    private static final long serialVersionUID = 5352714439194578175L;

    public RemoveVideoFromAlbum(@NotNull final String uri) {
        mUri = uri;
    }

    @NotNull
    @SerializedName("uri")
    public String mUri;

    public String getUri() {
        return mUri;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) { return true; }
        if (o == null || !getClass().equals(o.getClass())) { return false; }

        final RemoveVideoFromAlbum that = (RemoveVideoFromAlbum) o;

        return mUri.equals(that.mUri);
    }

    @Override
    public int hashCode() {
        return mUri.hashCode();
    }

}

File: models/src/main/java/com/vimeo/networking2/params/RemoveVideoFromAlbum.kt
package com.vimeo.networking2.params

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * An object that is used to patch video addition updates to an Album.
 *
 * @param uri The URI of the video.
 */
@JsonClass(generateAdapter = true)
data class RemoveVideoFromAlbum(

    @Json(name = "uri")
    val uri: String
)
